src/sparseml/transformers/__init__.py

```diff
@@ -18,23 +18,86 @@
 
 # flake8: noqa
 
+import logging as _logging
+
+
 try:
     import transformers as _transformers
 
-    transformers_import_error = None
-except Exception as transformers_import_err:
-    transformers_import_error = transformers_import_err
+    _transformers_import_error = None
+except Exception as _transformers_import_err:
+    _transformers_import_error = _transformers_import_err
 
 
-def _check_transformers_install():
-    if transformers_import_error is None:
-        return
-    raise ImportError(
-        "No installation of transformers found. It is recommended to use the "
-        "sparseml fork of transformers which can be installed under "
-        "sparseml[transformers] or git+https://github.com/neuralmagic/transformers.git"
+_LOGGER = _logging.getLogger(__name__)
+
+
+def _install_transformers_and_deps():
+
+    import pip as _pip
+    import sparseml as _sparseml
+
+    transformers_branch = (
+        "master"
+        if not _sparseml.is_release
+        else f"release/{_sparseml.version_major_minor}"
+    )
+    transformers_requirement = (
+        "transformers @ git+https://github.com/neuralmagic/transformers.git"
+        f"@{transformers_branch}"
+    )
+
+    _pip.main(
+        [
+            "install",
+            transformers_requirement,
+            "datasets",
+            "sklearn",
+            "seqeval",
+        ]
     )
 
+    try:
```

Comment:
I think we might want to extend this try catch to encompass the earlier portion as well in the event that pip doesn't have correct user access or something along those lines. This way we can give a good error to the user for the direction to fix